subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1389 @@
+		net_if_carrier_off(data->iface);
+	}
+
+	atomic_clear_bit(&data->state, CDC_NCM_CLASS_SUSPENDED);
+
+	LOG_DBG("Configuration disabled");
+}
+
+static void usbd_cdc_ncm_suspended(struct usbd_class_data *const c_data)
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+
+	atomic_set_bit(&data->state, CDC_NCM_CLASS_SUSPENDED);
+}
+
+static void usbd_cdc_ncm_resumed(struct usbd_class_data *const c_data)
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+
+	atomic_clear_bit(&data->state, CDC_NCM_CLASS_SUSPENDED);
+}
+
+static int usbd_cdc_ncm_ctd(struct usbd_class_data *const c_data,
+			    const struct usb_setup_packet *const setup,
+			    const struct net_buf *const buf)
+{
+	if (setup->RequestType.recipient == USB_REQTYPE_RECIPIENT_INTERFACE) {
+		if (setup->bRequest == SET_ETHERNET_PACKET_FILTER) {
+			LOG_DBG("bRequest 0x%02x (%s) not implemented",
+				setup->bRequest, "SetPacketFilter");
+			return 0;
+		}
+
+		if (setup->bRequest == SET_NTB_INPUT_SIZE) {
+			LOG_DBG("bRequest 0x%02x (%s) not implemented",
+				setup->bRequest, "SetNtbInputSize");
+			return 0;
+		}
+
+		if (setup->bRequest == SET_NTB_FORMAT) {
+			LOG_DBG("bRequest 0x%02x (%s) not implemented",
+				setup->bRequest, "SetNtbFormat");
+			return 0;
+		}
+	}
+
+	LOG_DBG("bmRequestType 0x%02x bRequest 0x%02x unsupported",
+		setup->bmRequestType, setup->bRequest);
+	errno = -ENOTSUP;
+
+	return 0;
+}
+
+static int usbd_cdc_ncm_cth(struct usbd_class_data *const c_data,
+			    const struct usb_setup_packet *const setup,
+			    struct net_buf *const buf)
+{
+	int ret = 0;
+
+	LOG_DBG("%d: %d %d %d %d", setup->RequestType.type, setup->bRequest,
+		setup->wLength, setup->wIndex, setup->wValue);
+
+	if (setup->RequestType.type != USB_REQTYPE_TYPE_CLASS) {
+		ret = -ENOTSUP;
+		goto out;
+	}
+
+	switch (setup->bRequest) {
+	case GET_NTB_PARAMETERS: {
+		struct ntb_parameters ntb_params = {
+			.wLength = sys_cpu_to_le16(sizeof(struct ntb_parameters)),
+			.bmNtbFormatsSupported = sys_cpu_to_le16(NTB_FORMAT_SUPPORTED),
+			.dwNtbInMaxSize = sys_cpu_to_le32(CDC_NCM_SEND_NTB_MAX_SIZE),
+			.wNdbInDivisor = sys_cpu_to_le16(4),
+			.wNdbInPayloadRemainder = sys_cpu_to_le16(0),
+			.wNdbInAlignment = sys_cpu_to_le16(CDC_NCM_ALIGNMENT),
+			.wReserved = sys_cpu_to_le16(0),
+			.dwNtbOutMaxSize = sys_cpu_to_le32(CDC_NCM_RECV_NTB_MAX_SIZE),
+			.wNdbOutDivisor = sys_cpu_to_le16(4),
+			.wNdbOutPayloadRemainder = sys_cpu_to_le16(0),
+			.wNdbOutAlignment = sys_cpu_to_le16(CDC_NCM_ALIGNMENT),
+			.wNtbOutMaxDatagrams = sys_cpu_to_le16(CDC_NCM_RECV_MAX_DATAGRAMS_PER_NTB),
+		};
+
+		LOG_DBG("GET_NTB_PARAMETERS");
+		net_buf_add_mem(buf, &ntb_params, sizeof(ntb_params));
+		break;
+	}
+	case SET_NTB_FORMAT:
+		LOG_DBG("SET_NTB_FORMAT");
+		break;
+
+	case SET_ETHERNET_PACKET_FILTER:
+		LOG_DBG("SET_ETHERNET_PACKET_FILTER");
+		break;
```

Comment:
Missed that. Looks like typo, GET_NTB_FORMAT? (Needs to be implemented or removed if not required).

But there is no GET_ETHERNET_PACKET_FILTER.

(Set request, SET_.*, are handled above in the control-to-device requests. You will never get this kind of requests in control-to-host.)